### DISCUSSION TITLE
Does WPFUI support .NET 8.0?

### DISCUSSION THREAD

**Discussion by hl2guide** [lepoco/wpfui#898] [Q&A] [answered]
Does WPFUI officially support .NET 8.0?

I created a new project using 'WPF UI - Fluent Navigation' and made no edits yet.

I get errors even if I leave the solution's Target Framework as the default `.NET 7.0` or `.NET 8.0`:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	XDG-0001	The resource "TextFillColorPrimaryBrush" could not be resolved.	UiDesktopApp1	C:\Users\Dean\source\repos\UiDesktopApp1\Views\Windows\MainWindow.xaml	19	
Error	XDG-0001	The resource "ApplicationBackgroundBrush" could not be resolved.	UiDesktopApp1	C:\Users\Dean\source\repos\UiDesktopApp1\Views\Windows\MainWindow.xaml	16	
Error	XDG-0001	The resource "TextFillColorPrimaryBrush" could not be resolved.	UiDesktopApp1	C:\Users\Dean\source\repos\UiDesktopApp1\Views\Windows\MainWindow.xaml	17	
```

**Comment by pomianowski** [comment, marked as answer]
Preview versions support the .NET 8 SDK. As an example, Gallery app uses .NET 8
https://github.com/lepoco/wpfui/blob/7fcc2d8410f056d1d6de97c2f8078b3d8b38539b/src/Wpf.Ui.Gallery/Wpf.Ui.Gallery.csproj#L5

https://www.nuget.org/packages/WPF-UI/3.0.0-preview.13

**Comment by cbradley11** [comment]
I still get these errors in 3.0.4 targeting .NET 8.0. It's only TextFillColorPrimaryBrush and ApplicationBackgroundBrush and only in MainWindow.xaml.

**Reply by SuperBrain** [reply]
> I still get these errors in 3.0.4 targeting .NET 8.0. It's only TextFillColorPrimaryBrush and ApplicationBackgroundBrush and only in MainWindow.xaml.

I'm confirming the same issue, only present in `MainWindow.xaml`, while the same resources are properly resolved when referenced in any page.

This, however, does not affect the build process, I can run the application and hot-reload works fine. The only thing affected is the fact that design-time colors are not applied on MainWindow (yet they are on all pages).

EDIT:
Replacing `DynamicResource` with `ui:ThemeResource` for all affected keys fixed this issue for me, though whether this is correct approach I don't know and would like someone to clarify.
